roles/docker_build_tag_push/tasks/main.yaml

```diff
@@ -0,0 +1,129 @@
+---
+#
+# This role creates the docker webserver and database images, then tags and 
+# pushes the images to a private registry that is setup in another role.
+# 
+# Two files containing the are copied to the device under test.  Each Dockerfile
+# is modified specifically to pull down its own base OS, mariadb package, and
+# apache package.
+#    mariadb_cont.tgz
+#    web_cont_2.tgz
+#
+
+# Need to handle the cases of building the docker containers for each variant
+
+  - name: Fail if ansible_docker0.ipv4.address is not defined
+    fail:
+      msg: "ansible_docker0.ipv4.address it not defined!"
+    when: ansible_docker0.ipv4.address is not defined
+    run_once: true
+
+# Red Hat
+  - name: copy mariadb_cont.tgz
+    copy: 
+      src=roles/docker_build_tag_push/files/rhelah/mariadb_cont.tgz
+      dest=/root
+      owner=root
+      group=root
+      mode=0644
+    when: ansible_distribution == "RedHat"
+
+  - name: copy web_cont_2.tgz
+    copy: 
+      src=roles/docker_build_tag_push/files/rhelah/web_cont_2.tgz
+      dest=/root
+      owner=root
+      group=root
+      mode=0644
+    when: ansible_distribution == "RedHat"
+
+# Fedora 
+  - name: copy mariadb_cont.tgz
+    copy: 
+      src=roles/docker_build_tag_push/files/fedora/mariadb_cont.tgz
+      dest=/root
+      owner=root
+      group=root
+      mode=0644
+    when: ansible_distribution == "Fedora"
+
+  - name: copy web_cont_2.tgz
+    copy: 
+      src=roles/docker_build_tag_push/files/fedora/web_cont_2.tgz
+      dest=/root
+      owner=root
+      group=root
+      mode=0644
+    when: ansible_distribution == "Fedora"
+
+# CentOS
+  - name: copy mariadb_cont.tgz
+    copy: 
+      src=roles/docker_build_tag_push/files/centos/mariadb_cont.tgz
+      dest=/root
+      owner=root
+      group=root
+      mode=0644
+    when: ansible_distribution == "CentOS"
+
+  - name: copy web_cont_2.tgz
+    copy: 
+      src=roles/docker_build_tag_push/files/centos/web_cont_2.tgz
+      dest=/root
+      owner=root
+      group=root
+      mode=0644
+    when: ansible_distribution == "CentOS"
+
+
+# Common build, tag, push
+  - name: untar db file
+    unarchive: src=/root/mariadb_cont.tgz dest=/root/ copy=no
+
+  - name: build db docker image
+    docker_image: path="/root/db" name="dbforweb" state=build
```

Comment:
Stylistically, I'd prefer to see the args split across lines for these two tasks:

```yaml
- name: untar db file
  unarchive: 
    src: /root/mariadb_cont.tgz 
    dest: /root/
    copy: no

- name: build db docker image
  docker_image: 
    path: "/root/db" 
    name: "dbforweb" 
    state: build
```